data_steward/cdr_cleaner/cleaning_rules/deid/explicit_identifier_suppression.py

```diff
@@ -0,0 +1,206 @@
+"""
+# Python imports
+import logging
+
+# Project imports
+from resources import get_concept_id_fields
+from utils import pipeline_logging
+from common import OBSERVATION, JINJA_ENV
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+
+LOGGER = logging.getLogger(__name__)
+
+ISSUE_NUMBERS = ['DC1363']
+EXPLICIT_IDENTIFIER_CONCEPTS = '_explicit_records_identifier_concepts'
+
+# Creates _explicit_identifier_concepts lookup table and populates with the concept_ids that are
+# from the concept table
+LOOKUP_TABLE_CREATION_QUERY = JINJA_ENV.from_string("""
+CREATE TABLE IF NOT EXISTS `{{project_id}}.{{sandbox_dataset}}.{{lookup_table}}` AS 
+(
+WITH
+  explicit_concept_ids AS (
+  SELECT
+    *
+  FROM
+    `{{project_id}}.{{dataset_id}}.concept`
+  WHERE
+    REGEXP_CONTAINS(concept_code, 'Signature') OR((REGEXP_CONTAINS(concept_code, 'Name')
+        AND REGEXP_CONTAINS(concept_code, r'(First)|(Last)|(Middle)|(Help)'))) OR((REGEXP_CONTAINS(concept_code, '(Address)|(PhoneNumber)')
+        AND NOT REGEXP_CONTAINS(concept_code, '(PIIState)|(State_)|(ZIP)'))
+      AND concept_class_id != 'Topic') OR(REGEXP_CONTAINS(concept_code, 'SecondaryContactInfo')
+      AND concept_class_id = 'Question') OR((REGEXP_CONTAINS(concept_code, 'SocialSecurity')
+        AND concept_class_id = 'Question')))
+SELECT
+   DISTINCT *
+FROM
+  explicit_concept_ids
+UNION DISTINCT
+SELECT
+  DISTINCT *
+FROM
+  `{{project_id}}.{{dataset_id}}.concept`
+WHERE
+  concept_id IN(
+  SELECT
+    cr.concept_id_2
+  FROM
+    explicit_concept_ids AS c
+  JOIN
+    `{{project_id}}.{{dataset_id}}.concept_relationship` AS cr
+  ON
+    c.concept_id = cr.concept_id_1
+  WHERE
+    cr.relationship_id = 'Maps to')
+)
+""")
+
+# Sandbox query to identify all the records with possible explicit record identifier concepts
+SANDBOX_EXPLICIT_IDENTIFIER_RECORDS = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` AS
+(
+SELECT
+      ob.*
+    FROM `{{project_id}}.{{dataset_id}}.{{observation_table}}` AS ob
+    {% for concept_field in concept_fields %}
+    LEFT JOIN `{{project_id}}.{{sandbox_dataset_id}}.{{lookup_table}}` AS s{{loop.index}}
+      ON ob.{{concept_field}} = s{{loop.index}}.concept_id 
+    {% endfor %}
+    WHERE COALESCE(
+    {% for concept_field in concept_fields %}
+        {% if loop.previtem is defined %}, {% else %}  {% endif %} s{{loop.index}}.concept_id
+    {% endfor %}) IS NOT NULL
+)
+""")
+
+SUPPRESS_EXPLICIT_IDENTIFIER_RECORDS = JINJA_ENV.from_string("""
+ DELETE FROM
+  `{{project_id}}.{{dataset_id}}.{{observation_table}}`
+WHERE
+observation_id 
+IN( SELECT
+    observation_id
+    FROM `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` )
+    """)
+
+
+class ExplicitIdentifierSuppression(BaseCleaningRule):
```

Comment:
nit: consider using the [abstract class](https://github.com/all-of-us/curation/blob/3db95db14ac7e8eeed2a26cf71eeafbc49673836/data_steward/cdr_cleaner/cleaning_rules/deid/concept_suppression.py#L141), because we are going to harmonize this anyway in the future, however, I will leave this to you as to what you want to do with it, I would be satisfied with the code as-is. It's better not to duplicate the same code and reuse as much existing code as possible.

Reply:
I will leave it like this for now, but in the future, I will definitely update this to the abstract class.